lib/appmap/util.rb

```diff
@@ -122,7 +122,7 @@ def select_rack_headers(env)
         # Apparently, it's following the CGI spec in doing so.
         # https://datatracker.ietf.org/doc/html/rfc3875#section-4.1.18
         matching_headers = env
-          .select { |k,v| k.start_with? 'HTTP_' }
+          .select { |k,v| k.to_s.start_with? 'HTTP_' }
```

Comment:
This bug would blow up the agent when a third party gem would put symbols in the request env. (datadog).